lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -732,66 +733,157 @@ class MetricsEndpointProvider(Object):
     def __init__(
```

Comment:
Consider adding a decorator to `__init__` that would modify the arguments as follows:
- relation would be added a default, if only one exists in metadata.yaml (you won't have access to `.meta` from the decorator)
- relation would **not** have a default value otherwise.

This way we earn static validation.

Reply:
But would anybody reading the code understand that?

Reply:
I think we should resist the temptation to use programming constructs unless they are really warranted and the additional code actually simplifies something. Creating a decorator just for a single use from this viewpoint is not useful.